Add language and otherlanguages key to DocumentMetadata (#2068)
latex3/latex2e: base/doc/ltnews43.tex

\subsection{Setting the language in \cs{DocumentMetadata}}

Until now \cs{DocumentMetadata} knew only the \texttt{lang} key to set the main language.

> The key was only used to set the language in the PDF catalog and the XMP-metadata and didn't affect the language of the document. 

Reviewer: ```suggestion
Until now \cs{DocumentMetadata} knew only the \texttt{lang} key to set the main language.
The key was only used to set the language in the PDF catalog and the XMP-metadata and didn't affect the language of the document as specified with \pkg{babel} or \pkg{polyglossia}. 
```

With this release we are now adding two more keys: \texttt{language} and \texttt{otherlanguages}. The \texttt{language} key takes as argument a language in BCP~47 format, so, e.g., \texttt{de-AT} or \texttt{fr} or \texttt{gsw-u-sd-chzh} (Swiss German as used in the Canton of Zurich). The value is stored in the document properties and can be retrieved (expandably) with \verb+\GetDocumentProperty{document/language}+. 

> The value of the \texttt{language} key is also used to set the language in the PDF metadata, so it is also a replacement for the \texttt{lang} key. 

Reviewer: it isn't documented what happens if both keys are used. lang overwrites language or ...?

Author: it depends on the order, the last one wins (for the /Lang).

Reviewer: in which case that seems to me a valuable info for users and should probably be mentioned in ltnews and not just as part of the code documentation

If a different (shorter or more generic) value is wanted in the metadata, an optional argument can be used: \verb+language=[de]{gsw-u-sd-chzh}+.

> The \texttt{language} key does \emph{not} automatically change the language setup of a document: it doesn't change the hyphenation patterns or the fix names and doesn't set class options. It also doesn't forces the loading of a language package. 

Reviewer: ```suggestion
Just like \texttt{lang}, the \texttt{language} key does \emph{not} automatically change the language setup of a document: it doesn't change the hyphenation patterns or the fix names and doesn't set class options. It also doesn't forces the loading of a language package. 
```

The expectation is that packages that need to know the document language, like babel or 

> polyglossia, will in future read the value and react accordingly. 

Reviewer: ```suggestion
The expectation is that in the future packages that need to know the document language, like \pkg{babel} or 
\pkg{polyglossia}, will read the value and react accordingly. 
```

> The \texttt{otherlanguages} key can be used to declare more languages that are used 

Reviewer: ```suggestion
The \texttt{otherlanguages} key can be used to declare further languages that are used in addition to the  \enquote{main} language
```
